### PR TITLE
sd-stub: report stub location in efi vars

### PR DESCRIPTION
**Pull request by poettering**
Let's report the UKI path via EFI var too, to mirror what we already do for sd-boot.

This is split out of #33512. That PR contains further additions to bootctl to make use of the new variables more comprehensively. And a testcase too.

### CODE REVIEW

File: man/systemd-stub.xml [hunk 1]
```diff
@@ -459,12 +459,16 @@
       <varlistentry>
```
Comment (yuwata) [inline man/systemd-stub.xml:459]: commit message: "not not necessary"

File: man/systemd-stub.xml [hunk 2]
```diff
@@ -496,6 +496,18 @@
         <xi:include href="version-info.xml" xpointer="v237"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>StubDevicePartUUID</varname></term>
+        <term><varname>StubImageIdentifier</varname></term>
+
+        <listitem><para>Similar to <varname>LoaderDevicePartUUID</varname> and
+        <varname>StubImageIdentifier</varname>, but indicates the location of the unified kernel image EFI
+        binary rather than the the location of the boot loader binary, regardless if booted via a boot loader
```
Comment (yuwata) [inline man/systemd-stub.xml]: typo: the the location

**Review by yuwata — APPROVED** [review]
Not a matter, but the first commit is also in #34347.

**Comment by poettering** [comment]
> Not a matter, but the first commit is also in #34347.

yeah, both PRs have later commits that touch the same code.

**Pull request merged**